### PR TITLE
Feature/nolivestream

### LINKED ISSUE #190: Remove livestream
Remove the concept of livestreaming from our backend

### LINKED ISSUE #192: Add blob storage functionality
Can mostly be copied from the FunderMaps repo.

### PR DESCRIPTION
**Pull request by tabeckers**
This PR removes the concept of a livestream from our backend.
All services in `Swabbr.Core.Interfaces.Services` have been completely cleaned up.

Fixes #190 
Fixes #192 

Note: The controllers will be cleaned up properly in a separate PR.
Note: The repositories and their abstractions will be cleaned up properly in a separate PR.

### CODE REVIEW

File: Swabbr.Api/Controllers/ReactionsController.cs [hunk 1]
```diff
@@ -191,103 +187,35 @@ public async Task<IActionResult> GetReactionCountForVlog([FromRoute] Guid vlogId
...
-            catch (EntityNotFoundException e)
-            {
-                logger.LogError(e.Message);
-                return Conflict(this.Error(ErrorCodes.EntityNotFound, "Could not find reaction"));
-            }
-            catch (ReactionStateException e)
-            {
-                logger.LogError(e.Message);
-                return Conflict(this.Error(ErrorCodes.InvalidOperation, "Reaction not in upload awaiting state"));
+                return Conflict(this.Error(ErrorCodes.EntityNotFound, "Could not find target vlog"));
```
Comment (yorickdewid) [inline Swabbr.Api/Controllers/ReactionsController.cs:218]: Maybe return `problem()` ?

File: Swabbr.Core/Enums/ReactionState.cs [hunk 2]
```diff
@@ -8,29 +8,10 @@ namespace Swabbr.Core.Enums
...
-        /// </summary>
-        [EnumMember(Value = "finished")]
-        Finished,
-
-        /// <summary>
-        ///     Transcoding the reaction failed.
-        /// </summary>
-        [EnumMember(Value = "failed")]
-        Failed,
+        [EnumMember(Value = "up_to_date")]
```
Comment (yorickdewid) [inline Swabbr.Core/Enums/ReactionState.cs:13]: Why the annotation? Is this related to Dapper?

File: Swabbr.Core/Notifications/NotificationAction.cs [hunk 3]
```diff
@@ -9,44 +9,39 @@
...
-        /// <summary>
-        ///     Indicates that a followed profile is live.
-        /// </summary>
-        FollowedProfileLive = 0,
-
         /// <summary>
         ///     Indicates a followed profile posted a new vlog.
         /// </summary>
-        FollowedProfileVlogPosted = 1,
+        FollowedProfileVlogPosted = 0,
```
Comment (yorickdewid) [inline Swabbr.Core/Notifications/NotificationAction.cs:15]: This not causing any issues on the client side?

File: Swabbr.Core/Services/HealthCheckService.cs [hunk 4]
```diff
@@ -10,28 +10,24 @@ namespace Swabbr.Core.Services
...
 
         /// <summary>
-        ///     Checks the livestream service, notification service and database.
+        ///     Checks the notification service and database.
         /// </summary>
         public async Task<bool> IsHealthyAsync()
-            => !await _livestreamService.IsServiceOnlineAsync().ConfigureAwait(false) ||
-                !await _notificationService.IsServiceOnlineAsync().ConfigureAwait(false) ||
-                !await _healthCheckRepository.IsAliveAsync().ConfigureAwait(false);
+            => !await _notificationService.IsServiceOnlineAsync().ConfigureAwait(false) ||
```
Comment (yorickdewid) [inline Swabbr.Core/Services/HealthCheckService.cs:30]: Split, 2 separate health checks

File: Swabbr.Core/Services/UserSelectionService.cs [hunk 5]
```diff
@@ -8,41 +8,44 @@
...
 
         /// <summary>
         /// Constructor for dependency injection.
         /// </summary>
-        public HashDistributionService(IOptions<SwabbrConfiguration> options)
+        public UserSelectionService(IUserService userService,
+            IOptions<SwabbrConfiguration> options)
         {
+            _userService = userService ?? throw new ArgumentNullException(nameof(userService));
             if (options == null || options.Value == null) { throw new ArgumentNullException(nameof(options)); }
```
Comment (yorickdewid) [inline Swabbr.Core/Services/UserSelectionService.cs:34]:
```suggestion
            if (options?.Value == null) { throw new ArgumentNullException(nameof(options)); }
```

File: Swabbr.Core/Services/UserSelectionService.cs [hunk 6]
```diff
@@ -8,41 +8,44 @@
...
         /// <param name="offset"><see cref="TimeSpan"/></param>
         /// <returns><see cref="SwabbrUser"/> collection</returns>
-        public IEnumerable<SwabbrUser> GetForMinute(IEnumerable<SwabbrUser> users, DateTimeOffset time, TimeSpan? offset = null)
+        public async Task<IEnumerable<SwabbrUser>> GetForMinuteAsync(DateTimeOffset time, TimeSpan? offset = null)
         {
-            if (users == null) { throw new ArgumentNullException(nameof(users)); }
-            if (time == null) { throw new ArgumentNullException(nameof(time)); }
+            var users = await _userService.GetAllVloggableUsersAsync().ConfigureAwait(false);
 
             var day = new DateTime(time.Year, time.Month, time.Day);
```
Comment (yorickdewid) [inline Swabbr.Core/Services/UserSelectionService.cs:50]: This is the point where you add a few lines of comments. What is happening here, how does the algo work? Maybe move the comment from below to here?

File: Swabbr.Core/Services/VlogRequestService.cs [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        }
+
+        /// <summary>
+        ///     Sends a vlog record request to each user that
+        ///     should get a request in the given <paramref name="minute"/>.
+        /// </summary>
+        /// <param name="minute">The minute to check.</param>
+        public async Task SendVlogRequestsForMinuteAsync(DateTimeOffset minute)
+        {
+            var selectedUsers = await _userSelectionService.GetForMinuteAsync(minute).ConfigureAwait(false);
```
Comment (yorickdewid) [inline Swabbr.Core/Services/VlogRequestService.cs:53]: Perfect opportunity for `await foreach`

File: Swabbr.Core/Services/VlogService.cs [hunk 8]
```diff
@@ -31,204 +33,237 @@ public class VlogService : IVlogService
...
-            scope.Complete();
+        /// <summary>
+        ///     Gets all vlogs that belong to a user including
+        ///     their thumbnail details.
+        /// </summary>
+        /// <param name="userId">The corresponding user.</param>
+        /// <returns>All vlogs belonging to the user.</returns>
+        public async Task<IEnumerable<VlogWithThumbnailDetails>> GetVlogsFromUserWithThumbnailsAsync(Guid userId)
+        {
+            // TODO Async enumerable
```
Comment (yorickdewid) [inline Swabbr.Core/Services/VlogService.cs:138]: Copy from FM

File: Swabbr.Core/Services/VlogService.cs [hunk 9]
```diff
@@ -31,204 +33,237 @@ public class VlogService : IVlogService
...
+        /// </summary>
+        /// <param name="vlogId">The vlog to like.</param>
+        /// <param name="userId">The user that likes the vlog.</param>
+        public async Task LikeAsync(Guid vlogId, Guid userId)
         {
-            vlogId.ThrowIfNullOrEmpty();
-            userId.ThrowIfNullOrEmpty();
+            // It's not allowed to like your own vlog
+            var vlog = await _vlogRepository.GetAsync(vlogId).ConfigureAwait(false);
+            if (vlog.UserId == userId)
```
Comment (yorickdewid) [inline Swabbr.Core/Services/VlogService.cs:196]: Or silently ignore

File: Swabbr.Core/Storage/StorageObject.cs [hunk 10]
```diff
@@ -0,0 +1,35 @@
+﻿namespace Swabbr.Core.Storage
+{
+    /// <summary>
+    ///     Storage object settings.
+    /// </summary>
+    public class StorageObject
```
Comment (yorickdewid) [inline Swabbr.Core/Storage/StorageObject.cs:6]: This was never finished in FM, most likely will change

**Review by yorickdewid — CHANGES_REQUESTED** [review]

File: Swabbr.Api/Controllers/ReactionsController.cs (hunk 1 above)
Comment (tabeckers) [reply Swabbr.Api/Controllers/ReactionsController.cs:218]: We need consistent error handling, created #195. This is one for a separate PR in which I also take the controllers into account.

File: Swabbr.Core/Enums/ReactionState.cs (hunk 2 above)
Comment (tabeckers) [reply Swabbr.Core/Enums/ReactionState.cs:13]: Yes

File: Swabbr.Core/Notifications/NotificationAction.cs (hunk 3 above)
Comment (tabeckers) [reply Swabbr.Core/Notifications/NotificationAction.cs:15]: No, the client currently maps from string. An integer was added in the previous PR (#179) which has not yet been used in the client side.

File: Swabbr.Core/Services/UserSelectionService.cs (hunk 6 above)
Comment (tabeckers) [reply Swabbr.Core/Services/UserSelectionService.cs:50]: Agreed

File: Swabbr.Core/Services/VlogRequestService.cs (hunk 7 above)
Comment (tabeckers) [reply Swabbr.Core/Services/VlogRequestService.cs]: Implemented up to the repository, which will be refactored without dapper.

File: Swabbr.Core/Storage/StorageObject.cs (hunk 10 above)
Comment (tabeckers) [reply Swabbr.Core/Storage/StorageObject.cs:6]: Noted, made issue #197

**Comment by tabeckers** [comment]
Processed all suggestions which lead to some minor additions.

**Comment by tabeckers** [comment]
The issue linking system failed to automatically close the referenced issues. Doing so manually.
